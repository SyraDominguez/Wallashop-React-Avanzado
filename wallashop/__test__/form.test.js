import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Form from '../src/components/form';

const mockStore = configureStore([]);
const store = mockStore({
  ads: [],
  tags: [],
});

describe('Form component', () => {
  it('should render correctly and match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Form />
        </BrowserRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
