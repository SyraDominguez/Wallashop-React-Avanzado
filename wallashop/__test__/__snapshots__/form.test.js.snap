// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <form
    class="newAdsPage"
  >
    <h3>
      Crear un nuevo anuncio
    </h3>
    <div>
      <label
        for="name"
      >
        Nombre:
      </label>
      <input
        id="name"
        placeholder="Filtra por Nombre"
        required=""
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="price"
      >
        Precio:
      </label>
      <input
        id="price"
        placeholder="Filtra por Precio"
        required=""
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        <input
          type="checkbox"
        />
        Incluir foto del artículo?
      </label>
    </div>
    <div
      class="photoPlaceholder"
    >
      <p>
        ¡No has incluido una foto!
      </p>
    </div>
    <div>
      <label>
        Tags:
      </label>
    </div>
    <div>
      <label>
        <input
          checked=""
          name="sale"
          type="radio"
          value="true"
        />
        Venta
      </label>
      <label>
        <input
          name="sale"
          type="radio"
          value="false"
        />
        Compra
      </label>
    </div>
    <button
      type="submit"
    >
      Crear anuncio
    </button>
    <button
      type="reset"
    >
      Borrar
    </button>
  </form>
</DocumentFragment>
`;
